### DISCUSSION TITLE
obok plugin for calibre not working

### DISCUSSION THREAD

**Discussion by aashcher** [apprenticeharper/DeDRM_tools#1520] [Q&A]
Hi,

Here is the error:
No userkeys found to decrypt books with. No point in proceeding.

Tried with both Calibre 4.x and 5.x. Used the Obok plugin from the master branch (Calibre indicates version 6.7.0)
I've installed the Kobo application through the website, logged in, and I am able to read the book which I am trying to convert. The book file is seen in the Kobo kepub folder.

Here is the log file for Calibre debug mode:

calibre 4.23 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 2.7.16+
Windows: ('10', '10.0.18362', 'SP0', u'Multiprocessor Free')
Interface language: ru
Successfully initialized third party plugins: Obok DeDRM (6, 7, 0)
calibre 4.23 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 2.7.16+
Windows: ('10', '10.0.18362', 'SP0', u'Multiprocessor Free')
Interface language: ru
Successfully initialized third party plugins: Obok DeDRM (6, 7, 0)
Turning on automatic hidpi scaling
devicePixelRatio: 2.0
logicalDpi: 96.0 x 96.0
physicalDpi: 78.9126213592 x 79.2832369942
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.34] splash screen shown
[0.34] Initializing db...
[0.34] db initialized
[0.34] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[1.53] main UI initialized...
[1.53] Hiding splash screen
[13.75] splash screen hidden
[13.75] Started up in 13.75 seconds with 1 books
Running Obok DeDRM v6.7.0
DEBUG:   16.7 didn't find device
Obok v4.0.0
Copyright В© 2012-2020 Physisticated et al.
c:\users\...\appdata\local\temp\tmptqhvpj
DEBUG:   16.7 got kobodir C:/Users/.../AppData/Local/Kobo/Kobo Desktop Edition
Error parsing Kobo plist: no legacy user key found.
Trouble retrieving keys with newer obok method.
Traceback (most recent call last):
  File "calibre_plugins.obok_dedrm.action", line 121, in launchObok
  File "calibre_plugins.obok_dedrm.obok.obok", line 413, in userkeys
  File "calibre_plugins.obok_dedrm.obok.obok", line 447, in __getmacaddrs
TypeError: __init__() got an unexpected keyword argument 'text'

Is there any way to process the book?

**Comment by ElleKayEm** [comment]
I've seen people having issues with Obok quite a bit lately.  Just checked and it is still working for me.  I'm guessing the problem is here:

> Error parsing Kobo plist: no legacy user key found.
Trouble retrieving keys with newer obok method.

Maybe I have a "legacy user key" so Obok doesn't have to worry about the new method.  I don't know.

A workaround would be downloading the Adobe DRM version from Kobo's website and using DeDRM instead.  You'd need Adobe Digital Editions 2.0.1 authorized with an Adobe ID on the same machine and user as calibre with DeDRM.  Open the ACSM file you get from Kobo in ADE.  Then add the downloaded EPUB to calibre.  Adobe EPUBs are working with both calibre 4.23/DeDRM 6.8.1 and calibre 5.x/DeDRM 7.0.3.

**Reply by aashcher** [reply]
Great, that worked, thank you!

**Reply by tbutler67** [reply]
> A workaround would be downloading the Adobe DRM version from Kobo's website and using DeDRM instead. You'd need Adobe Digital Editions 2.0.1 authorized with an Adobe ID on the same machine and user as calibre with DeDRM. Open the ACSM file you get from Kobo in ADE. Then add the downloaded EPUB to calibre. Adobe EPUBs are working with both calibre 4.23/DeDRM 6.8.1 and calibre 5.x/DeDRM 7.0.3.

Unfortunately, ADE  4.5.11 (Mac) is also not working. 

ADE is refusing to load the ACSM file from the Kobo website; when I try to open the ACSM file, I get "Error getting License. License Server Communication Problem: E_ADEPT_DOCUMENT_OPEN_ERROR" My guess is that some time ago, I had set up an Adobe ID with Sony's bookstore for a Sony e-reader (IIRC the PRS-505 generation, so about 12-13 years), and that old account is interfering with trying to load Kobo books. I've tried removing the authorization and re-authorizing ADE, same result.

**Reply by ElleKayEm** [reply]
I've used the same Adobe ID since the Sony days.  Did you also delete the activation.dat file?  See this page:  https://helpx.adobe.com/digital-editions/kb/activation-errors-adobe-digital-editions.html

**Reply by tbutler67** [reply]
Yes, I also removed the activation.dat file.

**Reply by ElleKayEm** [reply]
Hmm...Googling that error turns up some ideas.  But it seems not everyone can get it working.

**Reply by tbutler67** [reply]
After messing around with it a little while, I decided to go ahead and try to patch the plug-in manually with the code from https://github.com/apprenticeharper/DeDRM_tools/pull/1586 - after that patch, it's working fine.

**Reply by ElleKayEm** [reply]
Thanks for reporting back!

**Comment by niltojal** [comment]
Se você estiver usando Windows 10 com a segurança do windows ativada, pode ser que seguindo esses passos consiga resolver o seu problema.

É o seguinte galera sigam esse caminho e alguns problemas na hora de salvar vai ser resolvido;>>

Abrir o Windows defender >> Configurações de proteção contra Vírus e Ameaças >> Acesso a Pastas Controladas >>>

Permitir um App por meio do Acesso por pastas controladas ; Aqui você vai adicionar o app ou programa que está tentando salvar o arquivo ( Exemplo ; Photoshop, CorelDraw e outros) observação ; você tem que escolher o programa ou app, geralmente está dentro do Disco C: em programas e arquivos de programa, escolha o app, Abrir e Permitir. pronto.

**Reply by ElleKayEm** [reply]
Never had to do that on Windows 10 myself.
